runtime/gc_glue_java/ArrayletObjectModelBase.cpp

```diff
@@ -79,39 +82,40 @@ GC_ArrayletObjectModelBase::expandArrayletSubSpaceRange(MM_MemorySubSpace * subS
 	}
 }
 
-UDATA
-GC_ArrayletObjectModelBase::getSpineSizeWithoutHeader(ArrayLayout layout, UDATA numberArraylets, UDATA dataSize, bool alignData)
+uintptr_t
+GC_ArrayletObjectModelBase::getSpineSizeWithoutHeader(ArrayLayout layout, uintptr_t numberArraylets, uintptr_t dataSize, bool alignData)
 {
 	/* The spine consists of three (possibly empty) sections, not including the header:
 	 * 1. the alignment word - padding between arrayoid and inline-data
 	 * 2. the arrayoid - an array of pointers to the leaves
 	 * 3. in-line data
 	 * In hybrid specs, the spine may also include padding for a secondary size field in empty arrays
 	 */
-	UDATA const slotSize = J9GC_REFERENCE_SIZE(this);
+	uintptr_t const slotSize = J9GC_REFERENCE_SIZE(this);
 	MM_GCExtensionsBase* extensions = MM_GCExtensionsBase::getExtensions(_omrVM);
-	UDATA spineArrayoidSize = 0;
-	UDATA spinePaddingSize = 0;
+	uintptr_t spineArrayoidSize = 0;
+	uintptr_t spinePaddingSize = 0;
 	if (InlineContiguous != layout) {
 		if (0 != dataSize) {
 			/* not in-line, so there in an arrayoid */
 			spinePaddingSize = alignData ? (extensions->getObjectAlignmentInBytes() - slotSize) : 0;
 			spineArrayoidSize = numberArraylets * slotSize;
 		}
 	}
-	UDATA spineDataSize = 0;
+	bool isAllIndexableDataContiguousEnabled = extensions->indexableObjectModel.isVirtualLargeObjectHeapEnabled();
+
+	uintptr_t spineDataSize = 0;
 	if (InlineContiguous == layout) {
 		spineDataSize = dataSize; // All data in spine
-	} else if (Hybrid == layout) {
-#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
-		if (extensions->indexableObjectModel.isDoubleMappingEnabled()) {
+		if (isAllIndexableDataContiguousEnabled && (!extensions->indexableObjectModel.isArrayletDataAdjacentToHeader(dataSize))) {
```

Comment:
to avoid extra assigment to 0 (it's already preset to 0), we could invert this condition and set to 'dataSize'